### DISCUSSION TITLE
It is possible to pass a negative value via `page_size` in `CursorPagination`

### DISCUSSION THREAD

**Discussion by adamlitz** [encode/django-rest-framework#8277] [Potential Issue] [answered]
https://github.com/encode/django-rest-framework/blob/580bf45ccfd5c423a938729907d813f4862dca38/rest_framework/pagination.py#L607

When used against queryset, it will raise the `AssertionError`, because querysets don't support negative slicing. I think validation is missing if it's less than 0.

**Comment by lovelydinosaur** [comment, marked as answer]
So the preceding line `self.page_size = self.get_page_size(request)` always returns either a positive integer or `None`.

(Unless you've overridden `get_page_size` to behave differently.)

**Reply by adamlitz** [reply]
@tomchristie 
Indeed, I used my own implementation of `get_page_size` where I didn't check if `page_szie` is a negative number. Thank you for your answer. :)
